### ISSUE TITLE
"Invalid version" on `vue create` command (clean install)

### ISSUE THREAD

**Issue by TheDutchCoder** [vuejs/vue-cli#5891]
### Version
4.5.6

### Environment info
```
N/A
```

### Steps to reproduce
1. Globally install vue-cli
2. Run vue create [myprojectname]

### What is expected?
The project to be created

### What is actually happening?
A semver error is thrown:
```
 ✝  ~/projects  vue create myamplifyproject
 ERROR  TypeError: Invalid Version: ^4.5.6
TypeError: Invalid Version: ^4.5.6
    at new SemVer (/Users/reinierkaper/.config/yarn/global/node_modules/semver/semver.js:312:11)
    at compare (/Users/reinierkaper/.config/yarn/global/node_modules/semver/semver.js:585:10)
    at Function.gt (/Users/reinierkaper/.config/yarn/global/node_modules/semver/semver.js:614:10)
    at Object.exports.generateTitle (/Users/reinierkaper/.config/yarn/global/node_modules/@vue/cli/lib/util/clearConsole.js:17:29)
```

**Comment by TheDutchCoder** [comment]
Apparently there was a stray yarn config that clashes with vue-cli. Odd.
I removed it and now it works.

**Comment by TheDutchCoder** [comment]
That was premature, now there's an error when actually creating the project:
```
 ERROR  Error: spawn yarn ENOENT
Error: spawn yarn ENOENT
    at Process.ChildProcess._handle.onexit (internal/child_process.js:232:19)
    at onErrorNT (internal/child_process.js:407:16)
    at process._tickCallback (internal/process/next_tick.js:63:19)
```

**Comment by LinusBorg** [comment]
can you give us a little bit more information about your environment what is your node version what is your yarn version etc? 

Also could you provide a little bit more information when exactly the error is occurring? you wrote that it happens after trying to create a project with default options - a screenshot of the terminal,  showing which of the steps were already done or in progress before the error appeared would be helpful

**Comment by TheDutchCoder** [comment]
Okay, so I just figured out that vue-cli requires yarn.
If I install yarn globally through Homebrew the create function works fine.

That might be worth adding to the docs? I'm still a little confused why it would require yarn though.

**Comment by LinusBorg** [comment]
It does not require yarn. It work with both npm and yarn. And it checks which is available on your system and then chooses that. You can also force it to use one or the other with a flag. All of this *is* documented.

For some reason, it seems to have falsely "detected" yarn on your system - maybe from a prior install was was not completely removed?

**Comment by TheDutchCoder** [comment]
I uninstalled yarn completely and only started working when I reinstalled it through Homebrew.

Doesn't seem to make a ton of sense (I only use npm but had yarn installed from waaaay back apparently). Other people are also having this issue by the way, see: https://github.com/vuejs/vue-cli/issues/5690

**Comment by corepay** [comment]
Same errors here with npm global install.  using nvm do not want to cross-pollute node with brew-node.

**Comment by mstjepan28** [comment]
I had an issue when I deleted Yarn from windows that some files stayed in AppData\Local\Yarn. Deleting them solved the problem for me

**Comment by shauryadhadwal** [comment]
I also had the same issue. I am using Ubuntu.
I used vue-cli before, so there is a file create `~/.vuerc` which saves all your presets. 
In here, package manager was set as "yarn" which I no longer had. After deleting the file and using vue create,
I was prompted to select package manager.

Note: Before this, I installed yarn with which the error was resolved. But would recommend to use the above method.

**Comment by andersenmp** [comment]
> I also had the same issue. I am using Ubuntu.
> I used vue-cli before, so there is a file create `~/.vuerc` which saves all your presets.
> In here, package manager was set as "yarn" which I no longer had. After deleting the file and using vue create,
> I was prompted to select package manager. The issue was resolved post this.
> 
> Note: Before this, I installed yarn with which the error was resolved. But would recommend to use the above method.

I've got the same issue after removing yarn in Windows 10.  Package Manager was set to yarn on my ~/.vierc file

{
  "useTaobaoRegistry": false,
  "packageManager": "yarn"
}

**Comment by andres-k** [comment]
> I also had the same issue. I am using Ubuntu.
> I used vue-cli before, so there is a file create `~/.vuerc` which saves all your presets.
> In here, package manager was set as "yarn" which I no longer had. After deleting the file and using vue create,
> I was prompted to select package manager. The issue was resolved post this.
> 
> Note: Before this, I installed yarn with which the error was resolved. But would recommend to use the above method.

Thank you. I also ran into the same issue using ASDF for managing different node environments and this gave me a lot of headache. 

Deleting the `~/.vuerc` file solved the issue.

**Comment by stuheiss** [comment]
tl;dr If you previously installed yarn with brew run "brew unlink yarn && brew link yarn"

I had the same issue immediately after updating vue-cli. This apparently removed /usr/local/bin/yarn which is the symlink created by running "brew install yarn" and points to /usr/local/Cellar/yarn/1.22.10/bin/yarn. Fix is to recreate the symlink /usr/local/bin/yarn by running "brew unlink yarn && brew link yarn"

**Comment by aritro-rakshit** [comment]
For Windows users uninstall and reinstall yarn it will solve your issue.
npm uninstall yarn 
npm install yarn

**Comment by pointer** [comment]
@aritro-rakshit ...thanks

**Comment by binodnepali** [comment]
I also came across the same issue. I have npm and yarn both installed globally in my system. But the vue-cli (4.1.2) could not choose (or prompt me to choose) the package manager. I have fixed it by setting a flag `vue create -m npm project-name and it worked for me.

**Comment by zhuweiyou** [comment]
@andersenmp  Thank you! Your answer solved my problem.

**Comment by tako-sphynx** [comment]
> I also had the same issue. I am using Ubuntu. I used vue-cli before, so there is a file create `~/.vuerc` which saves all your presets. In here, package manager was set as "yarn" which I no longer had. After deleting the file and using vue create, I was prompted to select package manager. The issue was resolved post this.
> 
> Note: Before this, I installed yarn with which the error was resolved. But would recommend to use the above method.

Hi. I have the same error after `npm create some-project`. want to use your solution but can't find `~/.vuerc` file. Can you help me somehow?
